### PR TITLE
[IMP] account: Added in payment invoice related

### PR DESCRIPTION
**Pull request by luistorresm**
When I generate a payment directly from the invoice, this generate a
payment, and relate the invoice that is paid in that record, the same
when I use the option to pay many invoice from the invoice list view.

But when I generate the payment from the bank.statement, the payment
that is created with that process, do not assign the invoices that are
paid whit the statement.

Why I need the invoices related in all cases?

To the customer payments, is generated a CFDI document where are
indicated the invoices that are related with the payment, but when thw
process comes from bank.statement, this are not found.

The process to assign the invoices in the payment, do not affect Odoo
process, then We could have this records related.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by luistorresm** [comment]
@nhomar @julioserna


Are you agree with this PR?

Regards

**Comment by luistorresm** [comment]
@Julioserna @nhomar 

Are you agree with this?

**Comment by JulioSerna** [comment]
👍

**Comment by nhomar** [comment]
@qdp-odoo 

This is necessary to manage properly the payments on l10n_mx_edi, and from a functional standpoint we strongly beleive this belongs to the basecode of account, can you verify this, please?

File: addons/account/models/account_bank_statement.py [hunk 1]
```diff
@@ -869,17 +869,21 @@ def process_reconciliation(self, counterpart_aml_dicts=None, payment_aml_rec=Non
         new_aml_dicts = new_aml_dicts or []
 
         aml_obj = self.env['account.move.line']
+        invoice_obj = self.env['account.invoice']
```
Comment (gdgellatly) [inline addons/account/models/account_bank_statement.py]: Why?

**Comment by luistorresm** [comment]
Done @gdgellatly 

Regards

**Comment by qdp-odoo** [comment]
I understand the need, but I think we should implement that in another way: invoice_ids should be a computed m2m field, so that we're sure it's always correct, and it'll be also clearer from the dev/code point of view.

But that should target master... I'm creating a task internally for that, unless you want to carry on with that improvement, @luistorresm ?

**Comment by luistorresm** [comment]
Hello @qdp-odoo 

Have you considered making the change of your last message in the new Odoo version?

I do not create the PR with the improve.

CC: @nim-odoo 

Regards

**Comment by qdp-odoo** [comment]
yes, on payments we now have 2 m2m fields: invoice_ids (legacy one, that we kept for easiness and to avoid breaking too much things) and reconciled_invoice_ids (which is computed and therefor always correct)

You should use this second field in l10n_mx_edi

**Comment by luistorresm** [comment]
Thanks @qdp-odoo 

I will check that field.

Regards

**Pull request closed without merging**